### DISCUSSION TITLE
Building with Ninja always triggering full rebuild when using -fsycl option on Windows

### DISCUSSION THREAD

**Discussion by TomArnaez** [intel/llvm#14798] [Q&A] [answered]
When building with Ninja and the -fsycl compile option I'm getting a full rebuild everytime on Windows, is it possible to do partial rebuilds?

`cmake -G Ninja -B build -S . -DCMAKE_CXX_COMPILER=clang++`

```
cmake_minimum_required(VERSION 3.29)

project(ric LANGUAGES CXX)
set(CMAKE_CXX_STANDARD 17)

add_library(main src/test.hpp src/test.cpp)
target_compile_options(main PRIVATE -fsycl)
```

ninja output:
```
C:\dev\repos\ninja_clang_test\build>ninja -d explain
ninja explain: C:/Users/THOMAS~1/AppData/Local/Temp/test-header-0f011e.h has no in-edge and is missing
ninja explain: C:/Users/THOMAS~1/AppData/Local/Temp/test-header-0f011e.h is dirty
ninja explain: CMakeFiles/main.dir/src/test.cpp.obj is dirty
ninja explain: main.lib is dirty
[2/2] Linking CXX static library main.lib

C:\dev\repos\ninja_clang_test\build>ninja -t deps
CMakeFiles/main.dir/src/test.cpp.obj: #deps 3, deps mtime 7436946457430313 (VALID)
    C:/dev/repos/ninja_clang_test/src/test.cpp
    C:/Users/THOMAS~1/AppData/Local/Temp/test-header-036cb1.h
    C:/dev/repos/ninja_clang_test/src/test.hpp
```

**Comment by AlexeySachkov** [comment, marked as answer]
Hi @TomArnaez, thanks for your report.

`test-header-036cb1.h` is a special header file that we emit during device compilation and implicitly include into host compilation. It serves as an interface/mechanism to pass some data from device compilation to SYCL RT and host compilation (like kernel names for example).

I will tag @mdtoguchi here for his feedback, but my first reaction is that we should convert this into an issue and investigate if we can do something here. That temporary file clearly depends on `test.cpp`, so if we could somehow communicate that to the build system, then it should fix the issue.

I know that there is some ongoing effort to redesign this implementation completely and we may get rid of integration header completely, thus also solving this problem, but I doubt that it will come any time soon.

**Reply by mdtoguchi** [reply]
This is a known issue that is being worked on by @zahiraam.  There was some work done to improve how the integration footer is pulled in, causing a side effect of the header dependency not being properly filtered out when performing dependency generation.

**Reply by zahiraam** [reply]
This should be fixed with https://github.com/intel/llvm/pull/14933.

**Comment by zahiraam** [comment]
This issue should be fixed by https://github.com/intel/llvm/pull/14933. @TomArnaez can you please conform that's the case? Thanks.

**Reply by TomArnaez** [reply]
Can confirm it's fixed.

**Reply by zahiraam** [reply]
Thanks. Can the issue be closed then?
